lib/hocon/impl/token.rb

```diff
@@ -29,4 +30,9 @@ def to_s
       Hocon::Impl::TokenType.name(@token_type)
     end
   end
+
+  def ==(other)
```

Comment:
The more complex token types that subclass `Token` (they should all be internal classes in `Tokens.rb`) also override the `.equals()` method in java so that their values are compared, so those methods should be ported over to ruby as well.